tests/data/doctrine-dbal.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace DoctrineDbalTest;
+
+use Doctrine\DBAL\Driver\Connection;
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+    public function foo(Connection $conn)
+    {
+        $result = $conn->query('SELECT email, adaid, gesperrt, freigabe1u1 FROM ada');
+        assertType('Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<0, 4294967295>, 1: int<0, 4294967295>, gesperrt: int<-128, 127>, 2: int<-128, 127>, freigabe1u1: int<-128, 127>, 3: int<-128, 127>}>', $result);
+
+        $fetch = $result->fetchNumeric();
+        assertType('array{string, int<0, 4294967295>, int<-128, 127>, int<-128, 127>}', $fetch);
+
+        $fetch = $result->fetchAssociative();
+        assertType('array{email: string, adaid: int<0, 4294967295>, gesperrt: int<-128, 127>, freigabe1u1: int<-128, 127>}', $fetch);
+
+        $fetch = $result->fetchAllNumeric();
+        assertType('array<int, array{string, int<0, 4294967295>, int<-128, 127>, int<-128, 127>}>', $fetch);
+
+        $fetch = $result->fetchAllAssociative();
+        assertType('array<int, array{email: string, adaid: int<0, 4294967295>, gesperrt: int<-128, 127>, freigabe1u1: int<-128, 127>}>', $fetch);
```

Comment:
fetchAll* should perhaps return `list<array{}>` ? I imagine this is true for all drivers but who knows.

Reply:
phpstan does not yet support the list type natively. atm you can do this in phpdoc, but internally its still a array.

I thought about using a `positive-int` key.. not sure this holds?

Reply:
Ah interesting :) Yes I'd think positive-int is true, if list<> is true.. I'm sure it's 99% true and there's someone with a weird implementation somewhere that breaks it but probably this won't matter?

Reply:
yeah, I think we should go as strict as possible.

Reply:
be careful. `positive-int` means `>0`, which is wrong for lists

Reply:
Thx. We used int<0, max>